src/random_test.cpp

```diff
@@ -1567,16 +1573,19 @@ Table *Table::table_id(TABLE_TYPES type, int id, Thd1 *thd) {
   }
 
   /* if temporary table encrypt variable set create encrypt table */
-  static auto temp_table_encrypt =
-      get_result("select @@innodb_temp_tablespace_encrypt", thd);
+
+  auto row = mysql_fetch_row(
+      get_result("select @@innodb_temp_tablespace_encrypt", thd));
+  static std::string temp_table_encrypt = row[0];
```

Comment:
get that result of temp_table_enrypt  outside of this scope. so we just call get_result("select @@innodb_temp_tablespace_encrypt") once.